### PR TITLE
Introduction to Atomic Enterprise Platform - Background, Problem Statement, New Capabilities

### PR DESCRIPTION
**Pull request by markllama**
This PR adds a single document which introduces Atomic Enterprise in terms of the problems it means so solve.

The document provides a framework for understanding where Atomic Enterprise fits in the container service space.  It describes the existing layers and technologies, indicating the capabilities each provides and the boundaries and limitations. Finally it details the problems Atomic Enterprise means to address and the features it brings to enable the creation of an enterprise quality container management service.

This document serves as a beginner's introduction to "Why Choose Atomic Enterprise" as well as a touch-point for evaluating additional capabilities and where they should be placed in the container management stack.

Currently the capability set details are left open pending input from designers and developers. Additional capabilities can be added and detailed as needed.  Contributors are encouraged to comment and add content.

### CODE REVIEW

File: introduction.md [hunk 1]
```diff
@@ -0,0 +1,435 @@
...
+orchestration and management platform, based on a set of open-source
+container deployment and management projects, and extended to provide
+additional capabilities and services.
+
+Atomic Enterprise is meant to provide the administrators and users the
+means to create, deploy and manage complex applications in a container
+service environment.
+
+This document lays a foundation for understanding software containers,
+container management and the challanges and potential for large scale
```
Comment (miminar) [inline introduction.md]: challenges

File: introduction.md [hunk 2]
```diff
@@ -0,0 +1,435 @@
...
+
+## Preface - The Layers of Containers and Container Management
+
+The first section is a brief (?) recap of the functional structure of
+a generalized container deployment ecosystem. If you're familiar with
+the complete stack then you can skip to the
+[next section](#containers-and-orchestration-with-kubernetes)
+
+Software container applications are still a young technology (at least
+in terms of common use and availablility). The introduction of Docker
```
Comment (miminar) [inline introduction.md]: typo

File: introduction.md [hunk 3]
```diff
@@ -0,0 +1,435 @@
...
+made container development and execution accessible, and opened
+peoples' minds to the explore the possibilites that containers may
```
Comment (miminar) [inline introduction.md]: s/possibilites/possibilities/

File: introduction.md [hunk 4]
```diff
@@ -0,0 +1,435 @@
...
+offer. Since then the ideas and implimentations have come in a flood,
```
Comment (miminar) [inline introduction.md]: s/implimentations/implementations/

File: introduction.md [hunk 5]
```diff
@@ -0,0 +1,435 @@
...
+
+Orchestration is a blanket term for "making stuff happen automatically
+and in concert" (in the non-musical sense.) In terms of the scoping
+rings of a complete container management system, orchestration has
+three different responsibilities corresponding to three rings: facing
+in to the host, connecting hosts into a cluster, and facing out of the
+cluster to additional resources and to users.
+
+Each of the elements of container orchestration bridges a gap between
+the container and some resource that the container needs to do it's job.
```
Comment (miminar) [inline introduction.md]: s/it's/its/

File: introduction.md [hunk 6]
```diff
@@ -0,0 +1,435 @@
...
+
+#### Orchestration Layer 1: Host Resource Management
+
+The Host Resource Management function of the orchestration system
+bridges the boundaries between remote resources or non-existant host
```
Comment (miminar) [inline introduction.md]: s/existant/existent/

File: introduction.md [hunk 7]
```diff
@@ -0,0 +1,435 @@
...
+and will be assigned to a single container host. It also provides a
+means to create and maintain a specified number of replicas of each
+pod. Replicas will be placed on different nodes.
+
+### Resource Allocation
+
+Kubernetes allocation mechanism is fairly simple. It does not offer
+any way to partition the cluster or to assign pods to sub-sets of the
+cluster for resource or policy management purposes. A Kubernetes
+cluster is flat and uniform. Other than the constraint that
```
Comment (miminar) [inline introduction.md]: to be continued? :)

File: introduction.md [hunk 8]
```diff
@@ -0,0 +1,435 @@
...
+with the networking services that are available.
+
+### Publication and External Access
+
+Because Kubernetes is concerned mostly with creating and running
+containers within the cluster, it does not offer integrated mechanisms
+for publishing the services to external users. It can provide
+listeners on well known ports on one or more external facing hosts
+running special kube-proxy processes, but Kubernetes provides no means
+to make applications running in the cluster accessable to clients
```
Comment (miminar) [inline introduction.md]: s/accessable/accessible/

File: introduction.md [hunk 9]
```diff
@@ -0,0 +1,435 @@
...
+outside. These capabilities must be managed manually.
+
+### Network Storage
+
+On the nodes, kubernetes acts to bridge the boundaries between the
```
Comment (miminar) [inline introduction.md]: s/k/K/

File: introduction.md [hunk 10]
```diff
@@ -0,0 +1,435 @@
...
+host and the containers. By itself though, it can only make use of the
+resources that have been configured on the host. For example,
+Kubernetes has the capability to mount NFS filesystems onto the host
+and then into new containers, but only if the host has already been
+configured to act as an NFS client. [check this. Better example? MAL]
+
+In commercial cloud services Kubernetes does that the ability to reach
```
Comment (miminar) [inline introduction.md]: I don't get this.

File: introduction.md [hunk 11]
```diff
@@ -0,0 +1,435 @@
...
+### Software Defined Networking - Open vSwitch
+
+[Explain how AE adds networking capabilities both internally and on the boundary of the cluster]
+
+[Contributors?]
+
+### Inbound Access and Publication - HAProxy and integrated SkyDNS
+
+When applications and services run in a containerized environment, the
+container mangement system establishes the connections between the
```
Comment (miminar) [inline introduction.md]: s/mangement/management/

File: introduction.md [hunk 12]
```diff
@@ -0,0 +1,435 @@
...
+containers so that they can work as a unit. By itself, this is not
+sufficient to make these applications available to users outside the
+container service, the actual intended audience. Making hosted
+services accessable to users requires two additional related
```
Comment (miminar) [inline introduction.md]: s/accessable/accessible/

File: introduction.md [hunk 13]
```diff
@@ -0,0 +1,435 @@
...
+capabilities:
+
+1. The container service must establish a connection end-point (IP
+   address and port) on the network border of the service. 
+
+1. The container service must publish the existance of this end point
```
Comment (miminar) [inline introduction.md]: s/existance/existence/

File: introduction.md [hunk 14]
```diff
@@ -0,0 +1,435 @@
...
+
+## References
+
+- [Docker](http://www.docker.com)
+
+    Docker is a Linux software container system
+
+- [Kubernetes](http://www.github.com/GoogleCloudPlatform/kubernetes)
+
+    Kubernetes is a software container orchestration plaform
```
Comment (miminar) [inline introduction.md]: s/plaform/platform/

**Comment by miminar** [comment]
I like it. Good job. It will outshine tutorial itself in terms of depth and insight. Tutorial, which focuses more on getting things done, could refer to sections of this document for more background information. Also I'd add some link to `README.md`. But thes nitpicks can be solved in later PRs.

File: introduction.md (hunk 7 above)
Comment (markllama) [inline introduction.md]: i have to go back and look at the context and the narrative flow to see if this needs more, and what.

File: introduction.md (hunk 10 above)
Comment (markllama) [inline introduction.md]: Yeah.  Me neither.  gotta pick one idea.

**Comment by markllama** [comment]
@miminar I believe I've addressed all of your notes (_thank you_! for the spell-check and other edits!) I'm going to be away for a couple of weeks.  Feel free to merge and add to this. I don't feel comfortable merging my own changes.

**Comment by CowboysFan** [comment]
@markllama What is the goal of this doc, and where is it going to be published? The reason I ask is that the Atomic Enterprise docs will be developed in the openshift-docs repo on its own branch so that we can reuse that content. I'm not sure if this doc fits into our standard docs suite, but just thought I'd pass this along. Thanks.

**Comment by markllama** [comment]
Without this background, many people can't understand what containers are
let alone where Atomic enterprise or Openshift fit.  Even within the red
hat container community, most people cannot clearly articulate what the
components are and how they relate. This document is a primer for those who
need background to start working with containers and with red hat
offerings. It is a touchstone for those who work with containers already,
but who's focus keeps them from having complete perspective. If the scope
of Openshift documents does not have a fit for this, that's fine. Project
Atomic needs it as an introduction and scoping reference.

**Comment by miminar** [comment]
LGTM.

**Pull request merged**